pkg/aspect/query/query.go

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright © 2021 Aspect Build Systems Inc
+
+Not licensed for re-use.
+*/
+
+package query
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/url"
+	"regexp"
+	"strings"
+
+	"github.com/manifoldco/promptui"
+	"github.com/pkg/browser"
+	"github.com/spf13/cobra"
+
+	"aspect.build/cli/pkg/aspecterrors"
+	"aspect.build/cli/pkg/bazel"
+	"aspect.build/cli/pkg/ioutils"
+)
+
+type PresetQuery struct {
+	Name        string
+	Description string
+	Query       string
+}
+
+type Query struct {
+	ioutils.Streams
+
+	bzl           bazel.Bazel
+	isInteractive bool
+
+	Presets   []*PresetQuery
+	ShowGraph bool
+}
+
+func New(streams ioutils.Streams, bzl bazel.Bazel, isInteractive bool) *Query {
+	return &Query{
+		Streams:       streams,
+		bzl:           bzl,
+		isInteractive: isInteractive,
+	}
+}
+
+func (q *Query) Run(_ *cobra.Command, args []string) error {
+	presets := make(map[string]*PresetQuery)
+	var names []string
+	for _, p := range q.Presets {
+		presets[p.Name] = p
+		names = append(names, fmt.Sprintf("%s: %s", p.Name, p.Description))
+	}
+
+	var preset *PresetQuery
+	if len(args) == 0 {
+		selectQueryPrompt := &promptui.Select{
+			Label: "Select a preset query",
+			Items: names,
+		}
+
+		i, _, err := selectQueryPrompt.Run()
```

Comment:
Answering your question about how to test this, you would need to do some dependency injection here. This is not trivial in this case, but the best you can do is to have a `promptui.Select` factory that gets injected in the `Query` constructor and allows you to mock it during testing to return another mocked interface that gets `Run()` and you control both `i` and `err`. Let's pair on this and I show you how to do it.

Reply:
That would be great. Let me know when you have some time. No rush